repository/src/main/java/ru/karalionak/rest/query/OrderedViewQuery.java

```diff
@@ -0,0 +1,80 @@
+package ru.karalionak.rest.query;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class OrderedViewQuery {
+    private static final QueryHelper queryHelper = new QueryHelper();
+    private final StringBuilder query;
+    private final List<Object> values;
+
+    {
+        values = new ArrayList<>();
+        query = new StringBuilder("SELECT * FROM gift_certificates " +
+                "JOIN gift_certificate_tag gct on gift_certificates.id = gct.certificate_id " +
+                "JOIN tags on tags.id = gct.tag_id");
+    }
+
+    public OrderedViewQuery(StringBuilder query, List<Object> values) {
+        this.query.append(query);
+        this.values.addAll(values);
+    }
+
+    public String getQuery() {
+        return query.toString();
+    }
+
+    public List<Object> getValues() {
+        return values;
+    }
+
+    public static QueryBuilder builder(){
+        return new QueryBuilder();
+    }
+
+    public static class QueryBuilder{
+        private final StringBuilder query;
+        private final List<Object> values;
+        private boolean conditionFlag;
+
+        QueryBuilder(){
+            this.query = new StringBuilder();
+            this.values = new ArrayList<>();
+        }
+
+        public QueryBuilder filteringByTagName(String tagName){
+            if(!conditionFlag){
+                this.query.append(" WHERE ");
+                this.conditionFlag = true;
+
+            }else{
+                this.query.append(" OR ");
+            }
+            this.query.append("(tags.tag_name=?)");
+            this.values.add(tagName);
+            return this;
+        }
+
+        public QueryBuilder filteringByRegex(String regex){
```

Comment:
When I read 'regex', it's hard to understand what it really does. Please, use more obvious names. 'filterByNameOrDescription'